pkg/harvester/edit/kubevirt.io.virtualmachine/index.vue

```diff
@@ -328,25 +326,44 @@ export default {
     },
 
     restartVM() {
-      if ( this.mode === 'edit') {
-        const cloneDeepNewVM = clone(this.value);
+      if (this.mode !== 'edit') {
+        return;
+      }
+      if (!this.value.isRunning) {
+        return;
+      }
+      const cloneDeepNewVM = clone(this.value);
 
-        delete cloneDeepNewVM?.metadata;
-        delete this.cloneVM?.metadata;
+      delete cloneDeepNewVM?.metadata;
+      delete this.cloneVM?.metadata;
+      delete this.cloneVM?.__clone;
 
-        const oldVM = JSON.parse(JSON.stringify(this.cloneVM));
-        const newVM = JSON.parse(JSON.stringify(cloneDeepNewVM));
+      const cpu = cloneDeepNewVM.spec.template.spec.domain.resources.limits.cpu;
 
-        if (!isEqual(oldVM, newVM) && this.isRestartImmediately) {
-          if (this.value.hasAction('restart')) {
-            this.value.doActionGrowl('restart', {});
-          }
+      cloneDeepNewVM.spec.template.spec.domain.resources.limits.cpu = cpu.toString();
```

Comment:
I don't think it should be dealt with here, it's weird

Reply:
I moved logical code to the mixin of harvester's vm.